packages/web-config-server/src/apiV1/dataBuilders/generic/compose/composeDataByDataPoint.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { fetchComposedData } from '/apiV1/dataBuilders/helpers';
+
+/**
+ * Configuration schema
+ * @typedef {Object} ComposeDataConfig
+ * @property {string} sortOrder
+ * @property {Object<string, { dataBuilder, dataBuilderConfig }>} dataBuilders
+ *
+ * Example config:
+ * {
+ *   "dataBuilders": {
+ *        builder1: {
+ *            dataBuilder: <some databuilder>,
+ *            dataBuilderConfig: <some config>,
+ *        },
+ *        builder2: {
+ *            dataBuilder: <some databuilder>,
+ *            dataBuilderConfig: <some config>,
+ *        }
+ *    }
+ * }
+ * 
+ * Data returned from fetchComposedData():
+ * {
+ *     builder1: { 
+ *        data: [{name: dataPoint1: value: '111'}, {name: dataPoint2: value: '222'}}
+ *     },
+ *     builder2: {
+ *        data: [{name: dataPoint1: value: '333'}, {name: dataPoint2: value: '444'}},
+ *     }
+ * }
+ * 
+ * The above will then be converted to the below and returned:
+ * { 
+ *    data: 
+ *        [
+ *          { name: dataPoint1, builder1: '111', builder2: '333' }, 
+ *          { name: dataPoint2, builder1: '222', builder2: '444' }, 
+ *        ]
+ *    }
+ */
+
+export const composeDataByDataPoint = async (config, aggregator, dhisApi) => {
```

Comment:
To keep this consistent with other names, may I suggest `composeDataByDataClass`? Although that is already defined as a helper, so happy to keep it as is, or see if you can somehow replace `dataPoint` with `dataClass`.

Reply:
I agree with you, is `composeDataPerDataClass` okay?

Reply:
Yep, that sounds good (although heaven help anyone trying to tell the difference between that and `composeDataByDataClass` 😂. I think it's the best that you could do though with the existing code).